### PR TITLE
[log] integrate logger with ECS

### PR DESCRIPTION
**Pull request by dustinxie**

### CODE REVIEW

File: blocksync/buffer.go [hunk 1]
```diff
@@ -78,15 +75,12 @@ func (b *blockBuffer) Flush(blk *block.Block) (bool, bCheckinResult) {
...
 		delete(b.blocks, heightToSync)
 		if err := commitBlock(b.bc, b.cs, blk); err != nil && errors.Cause(err) != blockchain.ErrInvalidTipHeight {
-			if errors.Cause(err) == poll.ErrProposedDelegatesLength || errors.Cause(err) == poll.ErrDelegatesNotAsExpected || errors.Cause(err) == db.ErrNotExist {
-				l.Debug("Failed to commit the block.", zap.Error(err), zap.Uint64("syncHeight", heightToSync))
-			} else {
-				l.Error("Failed to commit the block.", zap.Error(err), zap.Uint64("syncHeight", heightToSync))
-			}
+			l.With(
+				zap.Uint64("syncHeight", heightToSync),
+				zap.Strings("actionHash", blk.ActionHashs())).Error("Failed to commit the block.", zap.Error(err))
```
Comment (dustinxie) [inline blocksync/buffer.go:80]: this if{} is added in https://github.com/iotexproject/iotex-core/commit/2bf61cdd85789412055cd529f49a8db997321e99
looks like to suppress a specific non-critical issue in the past

verified with latest code, this error no longer happens, so remove it and use l.Error() for any error returned by commitBlock()

File: action/protocol/staking/protocol.go [hunk 2]
```diff
@@ -338,7 +339,9 @@ func (p *Protocol) handle(ctx context.Context, act action.Action, csm CandidateS
 	}
 
 	if receiptErr, ok := err.(ReceiptError); ok {
-		log.L().Debug("Non-critical error when processing staking action", zap.Error(err))
+		actionCtx := protocol.MustGetActionCtx(ctx)
```
Comment (raullenchai) [inline action/protocol/staking/protocol.go:342]: this could panic

File: action/protocol/staking/vote_reviser.go [hunk 3]
```diff
@@ -121,5 +121,11 @@ func (vr *VoteReviser) flush(height uint64, csm CandidateStateManager) error {
 			return err
 		}
 	}
+	log.L().Info("committed revise action",
+		zap.Uint64("height", height), zap.Int("number of cands", len(cands)))
+	for i := range cands {
```
Comment (raullenchai) [inline action/protocol/staking/vote_reviser.go:126]: why not log in the for-loop above?

**Review by CoderZhi — APPROVED** [review]

File: blockchain/block/block.go [hunk 4]
```diff
@@ -152,3 +153,13 @@ func (b *Block) TransactionLog() *BlkTransactionLog {
 	}
 	return &blkLog
 }
+
+// ActionHashs returns action hashs in the block
+func (b *Block) ActionHashs() []string {
```
Comment (raullenchai) [inline blockchain/block/block.go:158]: This doesn't below to `Block`

File: blocksync/buffer.go (hunk 1 above)
Comment (raullenchai) [reply blocksync/buffer.go:80]: Suggest do not touch this snippet of code if not necessary. You didn't see it perhaps b/c it is `Debug` log which is not enabled.

File: action/protocol/staking/protocol.go (hunk 2 above)
Comment (dustinxie) [reply action/protocol/staking/protocol.go:342]: ctx is passed from above with action context
MustGetActionCtx(ctx) is also called in downstream funcs like handleCreateStake(), it should not panic

File: action/protocol/staking/vote_reviser.go (hunk 3 above)
Comment (dustinxie) [reply action/protocol/staking/vote_reviser.go]: fixed

**Pull request merged**